source/guide_libreoffice.rst

```diff
@@ -0,0 +1,444 @@
+
+In the following, LibreOffice refers to any of the three distributions.
+A listing of the ways to get LibreOffice Online is available on the
+`official download page <LOOL_>`_ by The Document Foundation.
+
+Assume you chose `collabora/code`_. Pull the image from dockerhub and create
+a container:
+
+::
+  
+  [isabell@stardust ~]$ udocker pull collabora/code
+  Downloading layer: sha256:84ed7d2f608f8a65d944b40132a0333069302d24e9e51a6d6b338888e8fd0a6b
+  Downloading layer: sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
+  [...]
+  Downloading layer: sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
+  [isabell@stardust ~]$ udocker create --name=collabora-code collabora/code
+  c974f5ce-60c2-3643-8405-30b31e67165b
+  [isabell@stardust ~]$ 
+
+.. note::
+
+  The output of the second command is the UUID of the container named
+  ``collabora-code``. If you leave out the ``--name`` flag, you must later use
+  the UUID instead.
+
+
+Configuration
+=============
+
+First, configure `udocker`_ to use `fakechroot`_ instead of `proot`_:
+::
+
+  [isabell@stardust ~]$ udocker setup --execmode=F1 collabora-code
+  [isabell@stardust ~]$
+
+.. note::
+
+  By default, `udocker`_ uses `proot`_ to fake `chroot`_. However, `proot`_ does
+  not support real multi-threading and leads to a noticeable lag when editing
+  documents. An overview over the execmodes of udocker can be found in the
+  `udocker manual <https://github.com/indigo-dc/udocker/blob/master/doc/user_manual.md#327-setup>`_.
+
+Configuration file
+------------------
+
+To proceed, create a configuration directory, and copy the configuration file
+from the container:
+
+::
+  
+  [isabell@stardust ~]$ mkdir -p $HOME/etc/libreoffice/
+  [isabell@stardust ~]$ cp $HOME/.udocker/containers/collabora-code/ROOT/etc/loolwsd/loolwsd.xml $HOME/etc/libreoffice/
+  [isabell@stardust ~]$ chmod 0600 $HOME/etc/libreoffice/loolwsd.xml
+  [isabell@stardust ~]$
+
+Most options can be set in the configuration file, but some only on the
+commandline (see below). In ``$HOME/etc/libreoffice/loolwsd.xml`` you can see
+the possible config file options with their descriptions and default values.
+Some options can only be set on the If you change the configuration file while
+LibreOffice is running, it shuts down and gets restarted by supervisord, if
+configured right.
+
+.. note::
+
+  * The permissions are set because ``loolwsd.xml`` will contain the admin console password.
+  * The option ``user_interface.mode=classic|notebookbar`` is present in the config file but
+    for the docker image `collabora/code`_ must be set on the commandline.
+  * Exhaustive information about the commandline and environment options is
+    available in the `docker setup instructions
+    <https://www.collaboraoffice.com/code/docker/>`_ by Collabora.
+
+Commandline configuration
+-------------------------
+
+Now create a startup script ``$HOME/etc/libreoffice/run`` with the following content:
```

Comment:
The `etc/` directory is for config files. What do you think about placing the script int `$HOME/bin/run_libreoffice`?

Reply:
I put it there because it is wrapper containing configuration. *Maybe* I could put all config into the config file, even if some env vars are used in the official documentation.

Reply:
If you can get working without a wrapper script, that would be nice. Keep in mind that you can also set env vars using a supervisord `environment=` line. http://supervisord.org/configuration.html

Reply:
Maybe I can remove the config part from the wrapper script. I think it makes sense here to have a wrapper script and don't put any config in the supervisord file, because the udocker command requires quite some options and for debugging I found it helpful to be able to run the container unsupervised.